docs/upgradeFromD8toD9.md

```diff
@@ -0,0 +1,209 @@
+As a good practice we append the **current date **(YEAR-DAY-MONTH) to the filename. Here we assume today is September 1st of 2021.
+
+```Shell
+
+sudo tar -czvpf $HOME/archipelago-deployment-backup-20210109.tar.gz ../archipelago-deployment
+
+```
+
+The process may take a few minutes. Now let's verify all is there and the `tar.gz` is not corrupt.
+
+```Shell
+tar -tvvf $HOME/archipelago-deployment-backup-20210109.tar.gz 
+```
+
+You will see a listing of files. If corrupt (do you have enough space? did your ssh connection drop?) you will see:
+
+```
+tar: Unrecognized archive format
+```
+
+### Step 4:
+
+Restart your `docker-compose` ensemble and wait a little while for all to start. Newer versions of Minio may take up to 4 minutes to connect if you are using it as an AWS S3 Gateway.
+
+```Shell
+docker-compose up -d
+```
+
+## Upgrading
+
+### Step 1:
+
+Navigate to your current `archipelago-deployment` repo folder and open up the file `permissions` for some of your most protected Drupal files.
+
+```Shell
+sudo chmod 777 web/sites/default
+sudo chmod 666 web/sites/default/*settings.php
+sudo chmod 666 web/sites/default/*services.yml
+```
+
+### Step 2:
+
+We are going to tell `composer` to update `composer.json` requirements *but not* update yet all dependencies (means also your Drupal 8 is running this whole time without being affected, nice!). This is very important to avoid dead-locks of cross-dependant libraries.
+
+Run each line for D9 core and its base packages.
+
+```Shell
+docker exec -ti esmero-php bash -c "composer require drupal/core:^9 drupal/core-composer-scaffold:^9 drupal/core-project-message:^9 --update-with-dependencies --no-update"
+docker exec -ti esmero-php bash -c "composer require drupal/core-dev:^9 --update-with-dependencies --no-update"
+```
+
+Now we are going to update D8 Archipelago required and suggested modules into D9 requirements/compatible packages:
+
+```Shell
+docker exec -ti esmero-php bash -c "composer require drupal/form_mode_manager:^2.0@RC --update-with-dependencies --no-update"
+docker exec -ti esmero-php bash -c "composer require drupal/display_field_copy:^2.0 --update-with-dependencies --no-update"
+docker exec -ti esmero-php bash -c "composer require symfony/event-dispatcher:^4 --update-with-dependencies --no-update"
+docker exec -ti esmero-php bash -c "composer require symfony/yaml:^4  --update-with-dependencies --no-update"
+docker exec -ti esmero-php bash -c "composer require drupal/webform:^6  --update-with-dependencies --no-update"
+docker exec -ti esmero-php bash -c "composer require drupal/google_api_client:3.2 --update-with-dependencies --no-update"
+docker exec -ti esmero-php bash -c "composer require drupal/bootstrap_barrio:5.1 --update-with-dependencies --no-update"
+docker exec -ti esmero-php bash -c "composer require 'drupal/role_based_theme_switcher:^9' --update-with-dependencies --no-update"
```

Comment:
@aksm can you check this part for me please? Pretty sure you have a few commands there to update the existing theme_switcher instead of this ugly beta module, means also removing docker exec -ti esmero-php bash -c "composer remove drupal/role_theme_switcher --no-update" and docker exec -ti esmero-php bash -c " drush pm-uninstall role_theme_switcher"